code/android/Gradr/app/src/main/java/com/se2/gradr/gradr/ProfileActivity.java

```diff
@@ -321,4 +332,19 @@ private void save()
         }
 
     }
+
+    private boolean validate() {
+        boolean isValid = true;
+        String url = mPictureView.getText().toString();
+
+        if (url.length() > 0)
+            isValid = Patterns.WEB_URL.matcher(url).matches();
```

Comment:
Any URL. I went under the assumption that the user could use any image hosting service. Same goes for the front-end validation on the web app.